### DISCUSSION TITLE
IKEv2 communication not working in port 500 when disabling socket-default and enabling socket-dynamic

### DISCUSSION THREAD

**Discussion by amdrsantos** [strongswan/strongswan#927] [Q&A]
When compiling strongswan with --disable-socket-default --enable-socket-dynamic, the IKEv2 communication it is working over other ports than 500, but not with port 500.

Please check the log in attachment
[socket-dynamic.log](https://github.com/strongswan/strongswan/files/8140968/socket-dynamic.log)

Can you advise?

**Comment by tobiasbrunner** [comment]
The plugin enables UDP decapsulation for all opened sockets, so it can't be used to send from/to port 500 as those packets don't have a non-ESP marker and are therefore processed as ESP packets by the kernel. Also, since the socket/port can't be switched if a NAT is detected or MOBIKE is enabled, using port 500 on either side doesn't work (also see [this section about custom server ports](https://wiki.strongswan.org/projects/strongswan/wiki/NatTraversal#Custom-Server-Ports)).
